x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/effected_policy_select/effected_policy_select.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  EuiFormRow,
+  EuiSelectable,
+  EuiSelectableProps,
+  EuiSwitch,
+  EuiSwitchProps,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EuiSelectableOption } from '@elastic/eui/src/components/selectable/selectable_option';
+import { PolicyData } from '../../../../../../../common/endpoint/types';
+
+interface OptionPolicyData {
+  policy: PolicyData;
+}
+
+type EffectedPolicyOption = EuiSelectableOption<OptionPolicyData>;
+
+export interface EffectedPolicySelection {
+  isGlobal: boolean;
+  selected: PolicyData[];
```

Comment:
I wonder if we use the whole entity somewhere? I think I only saw extracting ids, so maybe keeping just list of strings could reduce some logic around the place.

Think of an editing case - you will most likely receive only list of ids (maybe additionally policy name, but I would think not the whole policy). Hence you will have to do all sorts of fancy logic to pass in that list.

Reply:
I prefer for it to "echo" back the set of options that were selected based on their input `options` array. Its just easier in my mind. Also, I specifically made this Component not be dependent on the TA implementation because I would imagine that when we get around to Exceptions, it would also be used there.

As for the data - we still need to retrieve all of the Policies for this component reglardless of whatever enrichment our API might do. That enrichment will likely be most useful on display cases (grid card ++ table)

Reply:
The things is that it is a property and you won't be able to pass it right after loading trusted app in edit view, because you'll just get ids. It just feels like you'll have to do more complicated logic later to facilitate this interface. But we can think of it when implementing edit view.

Reply:
Sounds good. I'm not seeing the complexity, but we'll address it when the time comes if needed